Library/Homebrew/extend/pathname.rb

```diff
@@ -436,11 +436,17 @@ def env_script_all_files(dst, env)
   end
 
   # Writes an exec script that invokes a java jar
-  def write_jar_script(target_jar, script_name, java_opts = "")
+  def write_jar_script(target_jar, script_name, java_opts = "", java_version: nil)
     mkpath
+    java_home = if java_version
+      <<~EOS
+        export JAVA_HOME="$(#{Language::Java.java_home_cmd(java_version)})"
+
+      EOS
+    end
     join(script_name).write <<~EOS
       #!/bin/bash
-      exec java #{java_opts} -jar #{target_jar} "$@"
+      #{java_home}exec java #{java_opts} -jar #{target_jar} "$@"
```

Comment:
Utter pedantry I realise (sorry) but does this not end up with yucky indentation in the output file?

Reply:
I don't think so? Unless you mean the two spaces between "java" and "-jar" when "java_opts" is empty?

Reply:
@apjanke np; when it was using `EOS` I think the beginning of this line would have had weird indentation.

Reply:
I actually think it wouldn't've, because of `~EOS`'s de-indentation. (That's the sort of thing that would bother me too, no worries on the pedantry.) But... moot point now. ;)